build-tools/packages/build-cli/src/commands/release/prepare.ts

```diff
@@ -0,0 +1,266 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { MonoRepo, Package } from "@fluidframework/build-tools";
+import async from "async";
+import chalk from "chalk";
+import execa from "execa";
+import { ResetMode } from "simple-git";
+
+import { findPackageOrReleaseGroup, packageOrReleaseGroupArg } from "../../args.js";
+import {
+	BaseCommand,
+	type Context,
+	Repository,
+	getPreReleaseDependencies,
+} from "../../library/index.js";
+
+type CheckFunction = (
+	context: Context,
+	rgOrPkg: MonoRepo | Package,
+	print?: boolean,
+) => Promise<CheckResult>;
+
+interface CheckResult {
+	result: boolean;
+	message?: string;
+	fixCommand?: string;
+	fatal?: boolean;
+}
+
+async function checkNoLocalChanges(context: Context): Promise<CheckResult> {
+	const status = await context.gitRepo.getStatus();
+	if (status !== "") {
+		return {
+			result: false,
+			message: "There are some local changes. The branch should be clean.",
+			fixCommand: "git stash",
+			fatal: true,
```

Comment:
Seems of for this to be a fatal error. I'm assuming this means the subsequent checks don't run? Seems like we should be able to continue running other tests in this case? I'm just guessing about what this means due to the lack of docs on fatal above.